platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/capabilities/CapabilitiesLocalComponentIntegrationTest.groovy

```diff
@@ -76,5 +77,64 @@ class CapabilitiesLocalComponentIntegrationTest extends AbstractIntegrationSpec
         then:
         outputContains("Could not resolve project :.")
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/26377")
+    def "ResolvedVariantResults reported by ResolutionResult and ArtifactCollection have same capabilities when they are added to configuration in hierarchy"() {
+        settingsFile << "include 'producer'"
+        file("producer/build.gradle") << """
+            group="com.foo"
+
+            task zip(type: Zip) {
+                archiveFileName = "producer.zip"
+                destinationDirectory = layout.buildDirectory
+            }
+
+            configurations {
+                dependencyScope("api") {
+                    outgoing {
+                        capability("com.foo:producer:2.0")
+                        capability("org.bar:extra-capability:1.0")
```

Comment:
🤔 Does it make sense to move one of these capabilities into the consumable so that we show we're combining them?